### ISSUE TITLE
False positive `used_before_assignment` with assignment in `except` block and unconditional raise in `try` block

### ISSUE THREAD

**Issue by skshetry** [PyCQA/pylint#6070]
### Bug description

```python
# test.py
import traceback


def func():
    try:
        raise Exception("description")
    except Exception:  # pylint: disable=broad-except
        stack_trace = traceback.format_exc()

    assert stack_trace
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:14:11: E0601: Using variable 'stack_trace' before assignment (used-before-assignment)
```

### Expected behavior

No false positives

### Pylint version

```shell
pylint 2.13.4
astroid 2.11.2
Python 3.10.0 (default, Oct 17 2021, 20:19:40) [Clang 12.0.5 (clang-1205.0.22.9)]
```

**Comment by skshetry** [comment]
Thanks, @jacobtylerwalls for the reply. I do find the above error message confusing as the `try-catch` is exhaustive, but I agree with your points and the design decisions.

The above code was from our tests so it is definitely flawed. Closing the issue. Again, thankyou for the prompt reply. 🙂
